api/v1_projects.inc

```diff
@@ -271,6 +271,123 @@ function render_project_page_json($row)
     ];
 }
 
+// POST v1/projects/:projectid/rounds/:roundid
+// checkout a new page to proofread
+function api_v1_project_checkout($method, $data, $query_params)
+{
+    global $pguser;
+
+    $project = $data[":projectid"];
+    $required_roundid = $query_params["roundid"];
+    validate_user_can_work_in_project($project, $required_roundid);
+
+    try {
+        $lpage = get_available_page($project->projectid, $project->state, $pguser, $err);
+    } catch (ProjectPageException $exception) {
+        throw new BadRequest($exception->getMessage());
+    }
+
+    if (is_null($lpage)) {
+        throw new BadRequest($err);
+    }
+    return render_page_data($lpage);
+}
+
+// GET page text
+function api_v1_project_page_option($method, $data, $query_params)
+{
+    global $pguser;
+
+    $project = $data[":projectid"];
+    $required_roundid = $query_params["roundid"];
+    validate_user_can_work_in_project($project, $required_roundid);
+    $page_state = $query_params["page_state"];
+    try {
+        $projectid = $project->projectid;
+        $lpage = get_indicated_LPage($projectid, $project->state, $data[":pagename"], $page_state, 0);
+        if ($data[":pageoption"] == "original") {
+            $lpage->revertToOriginal();
```

Comment:
GETs should not modify anything.

Reply:
It doesn't change anything on the server. LPage is just a temporary construction.